src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionPolicy.cs

```diff
@@ -46,6 +53,45 @@ public virtual void Initialize(Transmitter transmitter)
             this.Transmitter = transmitter;
         }
 
+        // Calculates the time to wait before retrying in case of an error based on
+        // http://en.wikipedia.org/wiki/Exponential_backoff
+        public virtual TimeSpan GetBackOffTime(string headerValue)
+        {
+            TimeSpan retryAfterTimeSpan;
+            if (!this.TryParseRetryAfter(headerValue, out retryAfterTimeSpan))
+            {
+                double delayInSeconds;
+
+                if (this.ConsecutiveErrors <= 1)
+                {
+                    delayInSeconds = SlotDelayInSeconds;
+                }
+                else
+                {
+                    double backOffSlot = (Math.Pow(2, this.ConsecutiveErrors) - 1) / 2;
+                    var backOffDelay = this.random.Next(1, (int)Math.Min(backOffSlot * SlotDelayInSeconds, int.MaxValue));
```

Comment:
What does this give us:
(int)Math.Min(backOffSlot * SlotDelayInSeconds, int.MaxValue)
why not just backOffSlot * SlotDelayInSeconds
and also why do we need to randomize it?